File: ENC221-0087-2024 - Week 8.py

class Square:
    """Square class with side length"""
    
    def __init__(self, side):
        self.side = side
    
    def get_side(self):
        return self.side
    
    def get_area(self):
        return self.side * self.side
    
    def display_info(self):
        print(f"Square - Side: {self.side}")
        print(f"Area: {self.get_area()}")


class Cube(Square):
    """Cube class extends Square"""
    
    def __init__(self, side):
        super().__init__(side)
    
    def get_volume(self):
        return self.side ** 3
    
    def get_surface_area(self):
        return 6 * self.get_area()
    
    def display_info(self):
        print(f"Cube - Side: {self.get_side()}")
        print(f"Surface Area: {self.get_surface_area()}")
        print(f"Volume: {self.get_volume()}")


# Test the classes
if __name__ == "__main__":
    print("=== Shape Inheritance Test ===\n")
    
    # Create Square object
    square = Square(5.0)
    print("Square Test:")
    square.display_info()
    
    print("\n" + "=" * 40 + "\n")
    
    # Create Cube objects
    cube1 = Cube(4.0)
    print("Cube Test:")
    cube1.display_info()
    
    print("\n" + "=" * 40 + "\n")
    
    cube2 = Cube(10.0)
    print("Large Cube Test:")
    cube2.display_info()

File: ENC221-0087-2024 - Week 8.cpp

#include <iostream>
#include <string>
#include <cmath>
#include <iomanip>

using namespace std;

// Square class
class Square {
private:
    double side;

public:
    // Constructor
    Square(double side) : side(side) {}

    // Method to get side length
    double getSide() const { return side; }

    // Method to calculate area
    double getArea() const { return side * side; }

    // Display square information
    virtual void displayInfo() const {
        cout << "Square - Side: " << side << endl;
        cout << "Area: " << getArea() << endl;
    }
    
    // Virtual destructor
    virtual ~Square() {}
};

// Cube class extends Square
class Cube : public Square {
public:
    // Constructor
    Cube(double side) : Square(side) {}

    // Method to calculate volume
    double getVolume() const {
        double side = getSide();
        return pow(side, 3);
    }

    // Method to calculate surface area
    double getSurfaceArea() const {
        return 6 * getArea();
    }

    // Override display to show cube information
    void displayInfo() const override {
        cout << "Cube - Side: " << getSide() << endl;
        cout << "Surface Area: " << getSurfaceArea() << endl;
        cout << "Volume: " << getVolume() << endl;
    }
};

// Test the classes
int main() {
    cout << "=== Shape Inheritance Test ===" << endl << endl;

    // Create Square object
    Square square(5.0);
    cout << "Square Test:" << endl;
    square.displayInfo();

    cout << endl << string(40, '=') << endl << endl;

    // Create Cube objects
    Cube cube1(4.0);
    cout << "Cube Test:" << endl;
    cube1.displayInfo();

    cout << endl << string(40, '=') << endl << endl;

    Cube cube2(10.0);
    cout << "Large Cube Test:" << endl;
    cube2.displayInfo();

    return 0;
}